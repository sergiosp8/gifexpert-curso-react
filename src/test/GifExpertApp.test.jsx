import { render, screen, fireEvent } from '@testing-library/react'
import { GifExpertApp } from '../GifExpertApp'

describe('Prueba <GifExpertApp', () => {
  test('debe de mostrar el componente igual al snapshot', () => {
    const { container } = render(<GifExpertApp />)
    expect(container).toMatchSnapshot()
  })

  test('El estado debe de tener una categoría por defecto', () => {
    render(<GifExpertApp />)
    expect(screen.getByText('One Punch')).toBeTruthy()
  })

  test('Debe de agregar una nueva categoría', () => {
    render(<GifExpertApp />)
    screen.debug()
    const input = screen.getByRole('textbox')
    fireEvent.change(input, { target: { value: 'Dragon Ball' } })
    fireEvent.keyDown(input, { key: 'Enter', keyCode: 13 })
  })
})
