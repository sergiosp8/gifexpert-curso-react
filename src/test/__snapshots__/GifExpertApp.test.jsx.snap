// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba <GifExpertApp debe de mostrar el componente igual al snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <p>
    Loading...
  </p>
  <div
    class="card-grid"
  />
</div>
`;
